odl/solvers/functional/default_functionals.py

```diff
@@ -1647,15 +1678,23 @@ def __init__(self, *functionals):
         >>> f_sum = odl.solvers.SeparableSum(l1, 5)
         """
         # Make a power space if the second argument is an integer
-        if (len(functionals) == 2 and
-                isinstance(functionals[1], Integral)):
+        if (len(functionals) == 2 and isinstance(functionals[1], Integral)):
             functionals = [functionals[0]] * functionals[1]
+        for func1 in functionals:
```

Comment:
I guess
```Python
from functools import reduce
...
range = reduce((lambda x, func: x if func.range.contains_set(x) else func.range), functionals)
```
would be a more elegant way, but needs the import. Any comments?

Reply:
Now that I think about it, I guess this is the way to do it (because it doesn't need the import):
```Python
range = RealNumbers()
for func in functionals:
    range = func.range if func.range.contains(range) else range
```
(Or maybe set `range = None` in the beginning?)

Reply:
Or maybe it is reasonable to implement a method for the unification of sets? Something like
```Python
class Set {...
    def unified_with(self, other):
        """This is the default implementation for the unification of two sets
        It only supports equal sets
        """
        if type(self) == type(other):
            return other
        else:
            raise NotImplementedError('Cannot get the unified field for {} and'
                                      '{}'.format(self, other))
```
And the Number implementation
```Python
    def unified_with(self, other):
        """Returns the unified Field of `self` and `other`"""
        return other if other.contains_set(self) else self
```
That way new fields could easily implement their own method.
If we would end up doing so you might want to discuss the name...

Reply:
I'd suggest the simplest possible solution:
```python
range = RealNumbers()
for func in functionals:
    if func.range.contains(range):
        range = func.range
```